wrapper/src/test/java/integration/container/tests/TopologyQueryTests.java

```diff
@@ -0,0 +1,273 @@
+    stmt.executeQuery(query);
+    ResultSet rs = stmt.getResultSet();
+    int cols = rs.getMetaData().getColumnCount();
+    List<String> columnTypes = new ArrayList<>();
+    for (int i = 1; i <= cols; i++) {
+      columnTypes.add(rs.getMetaData().getColumnTypeName(i));
+    }
+    assertEquals(expectedTypes, columnTypes);
+    conn.close();
+  }
+
+  @TestTemplate
+  @ExtendWith(TestDriverProvider.class)
+  @EnableOnDatabaseEngineDeployment(DatabaseEngineDeployment.AURORA)
+  @EnableOnNumOfInstances(min = 2)
+  public void testAuroraTimestamp(TestDriver testDriver) throws SQLException, ParseException {
+    LOGGER.info(testDriver.toString());
+
+    final Properties props = ConnectionStringHelper.getDefaultPropertiesWithNoPlugins();
+    DriverHelper.setConnectTimeout(testDriver, props, 10, TimeUnit.SECONDS);
+    DriverHelper.setSocketTimeout(testDriver, props, 10, TimeUnit.SECONDS);
+
+    // Get second instance since first one has null timestamps
+    String url =
+        ConnectionStringHelper.getWrapperUrl(
+            testDriver,
+            TestEnvironment.getCurrent()
+                .getInfo()
+                .getDatabaseInfo()
+                .getInstances()
+                .get(1)
+                .getHost(),
+            TestEnvironment.getCurrent()
+                .getInfo()
+                .getDatabaseInfo()
+                .getInstances()
+                .get(1)
+                .getPort(),
+            TestEnvironment.getCurrent().getInfo().getDatabaseInfo().getDefaultDbName());
+    LOGGER.finest("Connecting to " + url);
+
+    String dbInstanceIdentifier = TestEnvironment.getCurrent().getInfo()
+        .getDatabaseInfo()
+        .getInstances()
+        .get(0)
+        .getInstanceId();
+    Region region = Region.of(TestEnvironment.getCurrent()
+        .getInfo()
+        .getAuroraRegion());
+
+    RdsClient client = RdsClient.builder()
+        .region(region)
+        .build();
+    // Update the cluster to force timestamp to appear
+    assertDoesNotThrow(() -> util.updateInstanceCertificateIdentifier(
+        dbInstanceIdentifier, "rds-ca-rsa4096-g1"));
+
+    client.close();
+
+    String query = null;
+    SimpleDateFormat format;
+    if (TestEnvironment.getCurrent().getCurrentDriver() == TestDriver.PG) {
+      query = AuroraPgDialect.getTopologyQuery();
+      format = new SimpleDateFormat("yyy-MM-dd HH:mm:ssX");
+      format.setTimeZone(TimeZone.getTimeZone("GMT"));
+    } else {
+      query = AuroraMysqlDialect.getTopologyQuery();
+      format = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss.SSSSSS");
+    }
+
+    final Connection conn = DriverManager.getConnection(url, props);
+    assertTrue(conn.isValid(5));
+    Statement stmt = conn.createStatement();
+    stmt.executeQuery(query);
+    ResultSet rs = stmt.getResultSet();
+
+    Date date = null;
+    while (rs.next()) {
+      if (rs.getString(5) != null) {
+        date = format.parse(rs.getString(5));
```

Comment:
We shouldn't rely on parsing timestamp. Parsing is locale dependent and the test might fail if database (or session) default settings changes.

Is it possible to call rs.getTime() or rs.getTimestamp()?
Is it possible to get metadata rs.getMetadata() and then check column type?

Reply:
The first test already checks all the column types of the query, including the timestamp. Should this test be removed then?